backend/projects/views.py

```diff
@@ -528,3 +530,27 @@ def project_script_detail(request, project_uuid, script_uuid):
         check_if_admin_of_project(request.user.id, project.uuid)
         script.delete()
         return JsonResponse({}, status=status.HTTP_204_NO_CONTENT)
+
+
+@swagger_auto_schema(
+    methods=["post"],
+    responses={200: openapi.Response("Updates a metric preference.")},
+    tags=["Metrics"],
+)
+@api_view(["POST"])
+@permission_classes([permissions.IsAuthenticated])
+def metrics(request):
+    data = JSONParser().parse(request)
+    project_id = data["project"]
+    new_metrics = data["metrics"]
+    metrics = MetricsPreferences.objects.filter(
```

Comment:
you can use the `update_or_create` function
https://docs.djangoproject.com/en/3.0/ref/models/querysets/#update-or-create